TsRandomizer/Randomisation/Gate.cs

```diff
@@ -1,14 +1,40 @@
-﻿using System.Linq;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using TsRandomizer.Extensions;
 
 namespace TsRandomizer.Randomisation
 {
-	abstract class Gate
+	abstract class Gate : IEquatable<Gate>
 	{
 		public abstract bool CanBeOpenedWith(Requirement obtainedRequirements);
 
 		public abstract bool Requires(Requirement requirementsToCheck);
 
+		private static List<Gate> requirementGates = new List<Gate>();
+		public static List<Gate> GetRequirementGates(Gate gate)
+        {
```

Comment:
Not at all sure what you are even trying to do here, i dont grasp the logic here
but i will say this, using reflection based on sub type to get some members is a bad design, if you need access to the sub games add an
public abstract Gate[] Gates here, so that all 3 times of gates have to implement it in the way that best suits them